src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const WebhookClient = require('../client/WebhookClient');
+const { ApplicationCommandOptionTypes, InteractionResponseTypes } = require('../util/Constants');
+const MessageFlags = require('../util/MessageFlags');
+
+/**
+ * Represents a command interaction.
+ * @extends {Interaction}
+ */
+class CommandInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The ID of the invoked application command
+     * @type {Snowflake}
+     */
+    this.commandID = data.data.id;
+
+    /**
+     * The name of the invoked application command
+     * @type {string}
+     */
+    this.commandName = data.data.name;
+
+    /**
+     * Whether the reply to this interaction has been deferred
+     * @type {boolean}
+     */
+    this.deferred = false;
+
+    /**
+     * The options passed to the command.
+     * @type {CommandInteractionOption[]}
+     */
+    this.options = data.data.options?.map(o => this.transformOption(o, data.data.resolved)) ?? [];
+
+    /**
+     * Whether this interaction has already been replied to
+     * @type {boolean}
+     */
+    this.replied = false;
+
+    /**
+     * An associated webhook client, can be used to create deferred replies
+     * @type {WebhookClient}
+     */
+    this.webhook = new WebhookClient(data.application_id, data.token);
```

Comment:
```suggestion
    this.webhook = new WebhookClient(data.application_id, data.token, this.client.options);
```
passes along the ClientOptions; I've had issues where I didn't do this and I was debugging it for a while because I forgot the WebhookClient is separate

also maybe rename this from `webhook` to `followup` or something?

Reply:
It's also used to edit/delete the interaction reply, so `webhook` is a more fitting name

Reply:
Maybe there could be like an alias method, `interaction.followup()` that just does `interaction.webhook.send()`